test/end2end_test.go

```diff
@@ -553,9 +568,19 @@ func (te *test) tearDown() {
 	if te.srv != nil {
 		te.srv.Stop()
 	}
-	if len(te.srvs) != 0 {
-		for _, s := range te.srvs {
-			s.Stop()
+	if te.hSrv != nil {
+		if hs, ok := te.hSrv.(*health.Server); ok {
```

Comment:
We can probably make `hSrv` type `*health.Server`. But that could cause many other changes.
It should be easier if we can get rid of `test.healthServer` and always use the standard health server.

Also, I believe calling `Shutdown()` for health server is optional. So it should be OK to not do this.

Reply:
There are some tests which need a custom health service implementation (like the test where we have a test server which takes an inordinate amount of time to respond to the Watch() rpc), and we test that the client side reportFunc is not called. This is one example, but there are a handful of tests which do require custom health service implementation.

Reply:
Removed the Shutdown() call.